src/Pos/Block/Logic/Internal.hs

```diff
@@ -111,8 +114,12 @@ type BlockApplyMode ssc m
 applyBlocksUnsafe
     :: forall ssc m . BlockApplyMode ssc m
     => OldestFirst NE (Blund ssc) -> Maybe PollModifier -> m ()
-applyBlocksUnsafe blunds0 pModifier =
-    reportingFatal version $
+applyBlocksUnsafe blunds0 pModifier = reportingFatal version $ do
+    when (length blunds /= length blunds0) $ assertionFailed $
```

Comment:
I don't like how it was done. This `spanSafe` is confusing. This function does something only if `blunds0 == blunds`. So `spanSafe` must return `(blunds0, [])`. So why not just check that all blocks have same epoch as the first one (head)? Also these names complicate understanding.